spec/PeriodicBackgroundSync-index.bs

```diff
@@ -92,3 +94,95 @@ Fetch requests within the <code>periodicsync</code> event while [=in the backgro
+
+The user agent MAY include factors such as user engagement with the origin to decide this time interval, allowing origins with high user engagement to update their web apps more often. In addition, the user agent SHOULD have per-origin and cross-origin caps on this interval, as recommended in [[#privacy]] and Resource Usage sections.
+
+The user agent may also decide to retry each failed periodicSync event, capped to a user agent defined <dfn>maximum number of retries</dfn>.
+
+A possible algorithm to calculate the [=time to fire=], <var>timeToFire</var> for |registration| would involve running these steps:
+1. If this is the <dfn>first attempt of firing</dfn> the periodicSync event:
+    1. Let <var>now</var> represent the current time.
+    1. Let <var>siteEngagementLevel</var> represent the level of engagement the user has with the origin. Calculate <var>siteEngagementPenalty</var>, which is a factor of |siteEngagementLevel|, and is inversely proportional to it. Let <var>minPeriodicSyncInterval</var> be a global cap on time intervals between periodicSync events. Define <var>effectiveGap</var> as |siteEngagementPenalty|*|minPeriodicSyncInterval|.
+    1. Let <var>minIntervalForOrigin</var> be the minimum time interval allowed by the user agent, per origin. Set <var>delayForOrigin</var> to the multiple of |effectiveGap| greater than or equal to |minIntervalForOrigin|.
+    1. Let <var>timeTillScheduledEventForOrigin</var> be the time when the next periodicSync event is scheduled for the origin, if any, null otherwise.
+    1. If |timeTillScheduledEventForOrigin| is not null:
+        1. If |timeTillScheduledEventForOrigin| - |delayForOrigin| is greater than or equal to |minIntervalForOrigin|, abort these substeps.
+        1. If |delayForOrigin| is less than or equal to |timeTillScheduledEventForOrigin|, set |delayForOrigin| to |timeTillScheduledEventForOrigin| and abort these substeps.
+        1. If |delayForOrigin| is less than or equal to |timeTillScheduledEventForOrigin| + |minIntervalForOrigin|, set |delayForOrigin| to |timeTillScheduledEventForOrigin| + |minIntervalForOrigin|.
+    1. Let <var>minIntervalAcrossOrigins</var> be the minimum interval allowed between periodicSync events across origins. Let <var>timeSinceLastPeriodicSync</var> be the time since a periodicSync event was fired for any origin, if any, null otherwise.
+    1. If |timeSinceLastPeriodicSync| is null, set |timeToFire| to |delayForOrigin| + |now|.
+    1. Else:
+        1. If |timeSinceLastPeriodicSync| is greater than equal to |minIntervalAcrossOrigins|, set |timeToFire| to |delayForOrigin| + |now|.
+        1. Else, set <var>timeTillNextAllowedPeriodicSync</var> to |minIntervalAcrossOrigins| - |timeSinceLastPeriodicSync|. Set |timeToFire| to the maximum of |delayForOrigin| + |now|, and |timeTillNextAllowedPeriodicSync| + |now|.
+    1. Set the [=time to fire=] of |registration| to |timeToFire|.
```

Comment:
|registrations|'s [=time to fire=]

Also, I'd scope [=time to fire=] with the registration dfn, so it is accessible by using [=registration/time to fire=]

Reply:
Re comment 1: done.
Re: In terms of how this is visible in the html, there's no difference. Also, if I do this, the same logic would apply to all other definitions under registration. Can you explain your reasoning? Are you worried these will be used outside of the spec?

Reply:
done. The spec is a lot less readable, so I'd still appreciate an example of how this improves maintainability.

Reply:
Same reason we use scopes and namespaces in code instead of having everything as a global variables. I'm sure you'll appreciate it if you have to amend sections of the spec a few months down the line :)